hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,329 @@
+        if sb_smoothness is None:
+            sb_smoothness = sb_size * 0.05
+        else:
+            if not isinstance(sb_smoothness, BaseSignal):
+                if isinstance(sb_smoothness, np.ndarray) and sb_smoothness.size > 1:
+                    sb_smoothness = BaseSignal(sb_smoothness).T
+                else:
+                    sb_smoothness = BaseSignal(sb_smoothness)
+
+        if sb_smoothness.axes_manager.navigation_size != self.axes_manager.navigation_size:
+            if sb_smoothness.axes_manager.navigation_size:
+                warnings.warn('Sideband smoothness dimensions do not match neither reference nor hologram dimensions.'
+                              'The reconstruction will be performed with the first value.')
+                sb_smoothness_temp = np.float64(sb_smoothness.inav[0].data)
+            else:  # sb_position navdim=0, therefore map function should not iterate it:
+                sb_smoothness_temp = np.float64(sb_smoothness.data)
+        else:
+            sb_smoothness_temp = sb_smoothness.deepcopy()
+
+        # Convert sideband size from 1/nm or mrad to pixels
+        if sb_unit == 'nm':
+            sb_size_temp = sb_size_temp / np.mean(self.f_sampling)
+            sb_smoothness = sb_smoothness / np.mean(self.f_sampling)
+        elif sb_unit == 'mrad':
+            try:
+                ht = self.metadata.Acquisition_instrument.TEM.beam_energy
+            except AttributeError:
+                ht = int(input('Enter beam energy in kV: '))
+                self.metadata.add_node('Acquisition_instrument')
+                self.metadata.Acquisition_instrument.add_node('TEM')
+                self.metadata.Acquisition_instrument.TEM.add_node('beam_energy')
+                self.metadata.Acquisition_instrument.TEM.beam_energy = ht
+            momentum = 2 * constants.m_e * constants.elementary_charge * ht * 1000 *\
+                       (1 + constants.elementary_charge * ht * 1000 / (2 * constants.m_e *
+                                                                       constants.c ** 2))
+            wavelength = constants.h / np.sqrt(momentum) * 1e9 # in nm
+            sb_size_temp = sb_size_temp / (1000 * wavelength * np.mean(self.f_sampling))
+            sb_smoothness_temp = sb_smoothness_temp / (1000 * wavelength * np.mean(
+                self.f_sampling))
+
+        # Find output shape:
+        if output_shape is None:
+            # if sb_size.axes_manager.navigation_size > 0: #  Future improvement will give a possibility to choose
+            #     output_shape = (np.int(sb_size.inav[0].data*2), np.int(sb_size.inav[0].data*2))
+            # else:
+            #     output_shape = (np.int(sb_size.data*2), np.int(sb_size.data*2))
+            output_shape = self.axes_manager.signal_shape
+
+        # Logging the reconstruction parameters if appropriate:
+        if log_parameters:
+            _logger.info('Sideband position in pixels: {}'.format(sb_position))
```

Comment:
Initially that was done for the cases if larger stacks of holograms are reconstructed to have a possibility not to "spam" the log if it is needed. At the moment I did it the way you suggested.